src/XrdOss/XrdOss.hh

```diff
@@ -447,6 +451,23 @@ const char     *getTID() {return tident;}
 
 virtual        ~XrdOssDF() {}
 
+//-----------------------------------------------------------------------------
+//! Return advanced error information about the last error encountered by
+//! this object.  Only invoked if the XrdOSS indicates it has feature XRDOSS_AERR
+//! The advanced error feature provides the ability for more error information
+//! than is possible from the typical error code (e.g., unstructured error
+//! messages).
+//!
+//! @return Error object associated with the last error.  Caller does not own
+//! the returned memory.  Lifetime of the returned object is only guaranteed
+//! until the next method call.  May return nullptr even if an error occurred.
+//-----------------------------------------------------------------------------
+
+#if XRDOSS_ERR
+virtual const XrdOucErrInfo *getError() const;
+#else
+const XrdOucErrInfo *getError() const;
+#endif
```

Comment:
I am not a fan of changing interface in this manner. I think it's better to have a clean patch and add it only in the next major version.

Reply:
Is it clear when the next major version is expected?  The level of ugly hacks would likely be different if it's immediately after 5.6 versus waiting until 2024...